plugins/hydra_rq_launcher/hydra_plugins/hydra_rq_launcher/_core.py

```diff
@@ -0,0 +1,146 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+import logging
+import os
+import time
+import uuid
+from pathlib import Path
+from typing import Any, Dict, List, Optional, Sequence
+
+import cloudpickle
+import hydra
+from hydra.core.config_loader import ConfigLoader
+from hydra.core.hydra_config import HydraConfig
+from hydra.core.singleton import Singleton
+from hydra.core.utils import (
+    JobReturn,
+    configure_log,
+    filter_overrides,
+    run_job,
+    setup_globals,
+)
+from hydra.types import TaskFunction
+from omegaconf import DictConfig, OmegaConf, open_dict
+from redis import Redis
+from rq import Queue
+
+from .rq_launcher import RQLauncher
+
+log = logging.getLogger(__name__)
+
+
+def execute_job(
+    idx: int,
+    overrides: Sequence[str],
+    config_loader: ConfigLoader,
+    config: DictConfig,
+    task_function: TaskFunction,
+    singleton_state: Dict[Any, Any],
+    redis_job_id: str,
+    work_dir: Optional[str] = None,
+) -> JobReturn:
+    """Calls `run_job`
+    """
+    if work_dir is not None:
+        os.chdir(work_dir)
```

Comment:
why are ou chdiring here?
run_job is supposed to handle it for the job.

Reply:
I took this out in a follow-up commit. Reason for originally including this is detailed in my comment below: https://github.com/facebookresearch/hydra/pull/683#issuecomment-644942327